api/src/main/java/ca/bc/gov/educ/studentdatacollection/api/rules/validationrules/impl/IndependentSchoolGraduateStudentRule.java

```diff
@@ -32,6 +35,18 @@ public IndependentSchoolGraduateStudentRule(ValidationRulesService validationRul
 
     @Override
     public boolean shouldExecute(StudentRuleData studentRuleData, List<SdcSchoolCollectionStudentValidationIssue> validationErrorsMap) {
+        log.debug("In shouldExecute of IndependentSchoolGraduateStudentRule-V49: for collectionType {} and sdcSchoolCollectionStudentID :: {}" , studentRuleData.getCollectionTypeCode(),
+                studentRuleData.getSdcSchoolCollectionStudentEntity().getSdcSchoolCollectionStudentID());
+
+        log.debug("In shouldExecute of IndependentSchoolGraduateStudentRule-V49: Condition returned  - {} for sdcSchoolCollectionStudentID :: {}" ,
+                CollectionTypeCodes.findByValue(studentRuleData.getCollectionTypeCode(), studentRuleData.getSchool().getSchoolCategoryCode()).isPresent()
```

Comment:
Could we extract this expression into a local variable for re-use in both the log and return type?